### PR TITLE
Adding pre-submit images build

### PR DESCRIPTION
**Pull request by radoslawc**
Dependency for pre-submit testing PRs:  https://github.com/nephio-project/nephio/issues/373

### CODE REVIEW

File: .prow.yaml [hunk 1]
```diff
@@ -77,3 +77,363 @@ presubmits:
       - image: nephio/gotests:1671656214984921088
         command:
         - "/usr/local/bin/lichen.sh"
+
+  - name: build-push-pre-image-ueransim-deploy-fn
```
Comment (efiacor) [inline .prow.yaml:81]: Is there any way in prow to define a job template that could be reused by all these?

Comment (radoslawc) [reply .prow.yaml:81]: @efiacor there is templating implemented, but extent of it doesn't cover everything we need. If you see other projects prow configuration suffers from the same problem being enormous. For main configuration one can split it to single file per job for example, for inrepoconfig (.prow.yaml) that's not an option.

Comment (efiacor) [reply .prow.yaml:81]: Yes, the closest I found is this, which is huge - https://github.com/aws/eks-distro-prow-jobs/blob/main/docs/prowjobs.md
Maybe we could use plank to define some common stuff but don't know prow well enough to be honest.

**Review by gvbalaji — COMMENTED** [review]
/approve

**Pull request merged**